subsys/net/lib/aws_fota/src/aws_fota.c

```diff
@@ -0,0 +1,222 @@
+#include <aws_jobs.h>
+#include <aws_fota.h>
+#include <json.h>
+#include "aws_fota_internal.h"
+
+
+static struct pollfd fds;
+
+/* Enum for tracking the job exectuion state */
+static enum execution_status execution_state;
+
+/* Allocated strings for checking topics */
+static u8_t notify_next_topic[NOTIFY_NEXT_TOPIC_LEN + 1];
+static u8_t job_id_update_topic[JOB_ID_UPDATE_TOPIC_MAX_LEN + 1];
+static char version[CONFIG_VERSION_STRING_MAX_LEN + 1];
+static u8_t payload_buf[CONFIG_AWS_IOT_JOBS_MESSAGE_SIZE];
+
+/* Topic for updating shadow topic with version number */
+#define UPDATE_DELTA_TOPIC AWS "%s/shadow/update"
+#define UPDATE_DELTA_TOPIC_LEN (AWS_LEN +\
+				CONFIG_CLIENT_ID_MAX_LEN +\
+				(sizeof(/shadow/update) - 1))
+#define SHADOW_STATE_UPDATE "{\"state\":{\"reported\":{\"nrfcloud__fota_v1__app_v\":%s}}}"
+static int update_device_shadow_version(struct mqtt_client *const client)
+{
+	struct mqtt_publish_param param;
+	char update_delta_topic[UPDATE_DELTA_TOPIC_LEN + 1];
+	u8_t shadow_update_payload[CONFIG_DEVICE_SHADOW_MESSAGE_BUFFER_SIZE];
+
+	int ret = snprintf(update_delta_topic, sizeof(update_delta_topic),
+		       UPDATE_DELTA_TOPIC, c->client_id.utf8);
+	u32_t update_delta_topic_len = ret + 1;
+
+	if (ret >= UPDATE_DELTA_TOPIC_LEN) {
+		return -ENOMEM;
+	} else if (ret < 0) {
+		return ret;
+	}
+
+	ret = snprintf(shadow_update_payload,
+		       CONFIG_DEVICE_SHADOW_MESSAGE_BUFFER_SIZE,
+		       SHADOW_STATE_UPDATE,
+		       version);
+	u32_t shadow_update_payload_len = ret + 1;
+
+	if (ret >= UPDATE_DELTA_TOPIC_LEN) {
+		return -ENOMEM;
+	} else if (ret < 0) {
+		return ret;
+	}
+
+	param.message.topic.qos = 1;
```

Comment:
`MQTT_QOS_1_AT_LEAST_ONCE` instead of magic number.

Reply:
Will fix.